tests/commands/create_project.py

```diff
@@ -0,0 +1,269 @@
+    call_command(
+        "create_project",
+        "foo0",
+        "--preset-mapping=gnu")
+    foo0 = Project.objects.get(code="foo0")
+    assert foo0.fullname == "Foo0"
+    assert foo0.checkstyle == "standard"
+    assert foo0.filetypes.first().name == "po"
+    assert foo0.source_language.code == "en"
+    assert foo0.disabled is False
+    assert foo0.report_email == ""
+
+    assert (
+        foo0.config["pootle_fs.translation_mapping"]
+        == dict(default=u'/po/<language_code>.<ext>'))
+    assert foo0.config["pootle_fs.fs_type"] == "localfs"
+    assert (
+        foo0.config["pootle_fs.fs_url"]
+        == ("{POOTLE_TRANSLATION_DIRECTORY}%s"
+            % foo0.code))
+
+
+@pytest.mark.cmd
+@pytest.mark.django_db
+def test_cmd_create_existing_project(settings):
+    call_command(
+        "create_project",
+        "foo0",
+        "--preset-mapping=gnu")
+
+    with pytest.raises(ValidationError):
+        call_command(
+            "create_project",
+            "foo0",
+            "--preset-mapping=gnu")
+
+
+@pytest.mark.cmd
+@pytest.mark.django_db
+def test_cmd_create_project_title(capfd):
+    call_command(
+        "create_project",
+        "foo1",
+        "--name=Bar1",
+        "--preset-mapping=gnu")
+    foo1 = Project.objects.get(code="foo1")
+    assert foo1.fullname == "Bar1"
+
+
+@pytest.mark.cmd
+@pytest.mark.django_db
+def test_cmd_create_project_checkstyle(capfd):
+    call_command(
+        "create_project",
+        "foo0",
+        "--checkstyle=minimal",
+        "--preset-mapping=gnu")
+    foo0 = Project.objects.get(code="foo0")
+    assert foo0.checkstyle == "minimal"
+
+    with pytest.raises(CommandError):
+        call_command(
+            "create_project",
+            "foo1",
+            "--preset-mapping=gnu",
+            "--checkstyle=DOESNOTEXIST")
+
+
+@pytest.mark.cmd
+@pytest.mark.django_db
+def test_cmd_create_project_filetypes(capfd):
+    call_command(
+        "create_project",
+        "foo0",
+        "--filetype=ts",
+        "--preset-mapping=gnu")
+    foo0 = Project.objects.get(code="foo0")
+    assert foo0.filetypes.first().name == "ts"
+
+    call_command(
+        "create_project",
+        "foo1",
+        "--filetype=ts",
+        "--preset-mapping=gnu")
+    foo1 = Project.objects.get(code="foo1")
+    assert foo1.filetypes.first().name == "ts"
+
+    with pytest.raises(CommandError):
+        call_command(
+            "create_project",
+            "foo2",
+            "--preset-mapping=gnu",
+            "--filetype=DOESNOTEXIST")
+
+
+@pytest.mark.cmd
+@pytest.mark.django_db
+def test_cmd_create_project_report_email(capfd):
+    call_command(
+        "create_project",
+        "foo0",
+        "--report-email=foo@bar.com",
+        "--preset-mapping=gnu")
+    foo0 = Project.objects.get(code="foo0")
+    assert foo0.report_email == "foo@bar.com"
+
+    with pytest.raises(CommandError):
+        # invalid email address
```

Comment:
Instead of trusting that `CommanError` here is because of invalid email address I would rather check that the error message in the exception contains such error. See second example in https://docs.pytest.org/en/latest/assert.html#assertions-about-expected-exceptions